workflows/seqqc.nf

```diff
@@ -176,7 +176,32 @@ workflow SEQQC {
 
 workflow.onComplete {
     if (params.email || params.email_on_fail) {
-        NfcoreTemplate.email(workflow, params, summary_params, projectDir, log, multiqc_report)
+        def email_params = NfcoreTemplate.get_email_params(workflow, params, summary_params, projectDir, log, multiqc_report)
+
+        // Note: this is a mild hack, because sendMail does not work with S3 URIs.
+        // First, we attempt to send our email regularly. If email_params.mqcFile is
+        // a local file path (ie you're running this pipeline on an HPC or locally),
+        // this will succeed. If it refers to an S3 object, this would fail.
+        // When it fails, we can forgo the main body of the email (mostly pipeline metadata)
+        // and try make MultiQC HTML (which already has some of the metadata), the
+        // body of our email.
+
+        try {
+            // Send email using Nextflow's built-in emailer:
+            // https://www.nextflow.io/docs/latest/mail.html#advanced-mail
+            sendMail (
+                to: email_params.to,
+                subject: email_params.subject,
+                body: email_params.email_html,
+                attach: email_params.mqcFile
+            )
+        } catch (Exception e) {
+            sendMail (
+                to: email_params.to,
+                subject: email_params.subject,
+                body: email_params.mqcFile.text
```

Comment:
so the html text itself is the body of the email instead of an attachment?

Reply:
yes